### DISCUSSION TITLE
"Failed to connect to datasource" - after upgrading from v0.118.0 to latest version

### DISCUSSION THREAD

**Discussion by gallickgunner** [heremaps/harp.gl#2144] [Q&A] [answered]
Hello, I just upgraded harpgl from 0.118. It seems the `OmvTileDecoder` got replaced or renamed by the `VectorTileDecoder`. I renamed all instances of it and now I'm using `@here/harp-vectortile-datasource`. But there are a few errors regarding loading of textureAtlas and failing to connect to data source. 

Here is the full screen.
![Untitled](https://user-images.githubusercontent.com/40718428/110625276-08b9d600-81c1-11eb-8b2f-5bfdf3ca438a.png)

The errors regarding `maki_icons` were there before that so they aren't related to the current issue I guess but I'd be glad to know what it's about.

How I load the data,
```
const dataSource = new VectorTileDataSource( {
      baseUrl: 'https://vector.hereapi.com/v2/vectortiles/base/mc',
      apiFormat: APIFormat.XYZOMV,
      // styleSetName: 'tilezen',
      authenticationCode: 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
      authenticationMethod: {
                method: AuthenticationMethod.QueryString,
                name: 'apikey',
            },
    } );
```

How the decoder.js looks like

```
import { VectorTileDecoderService } from '@here/harp-vectortile-datasource/index-worker';

VectorTileDecoderService.start();

```

Also the getting started guide hasn't been updated properly. It's still using `OmvTileDecoderService` in the decoder.

**Comment by nzjony** [comment, marked as answer]
@gallickgunner , thanks for your issue.

Do you have your repo somewhere I can take a look?

It seems that your style is a bit out of date, did you make some changes to it?

Regarding the documentation, I have made a quick fix: https://github.com/heremaps/harp.gl/pull/2128/

However long term, it would be good to have a central place to help people get up and started, we have: https://developer.here.com/tutorials/harpgl but also this repo... will need to think about it.

**Reply by gallickgunner** [reply]
Hmm, I'll try to create a minimalistic example since I faced this issue at work and can't share that code. By style you mean map theme? We are using a custom theme tho. The one passed in through the constructor of `MapView`

**Reply by gallickgunner** [reply]
@nzjony - Ok my bad. I think it was the faulty map style as you said. I used the default one and map loads now. However I'm getting a rate limit reached error when zooming in/out from the HERE api. My usage details in my account show 0.9% of the transactions are used. I'm not familiar with web dev and such so what's the issue here? I don't think we have exhausted the number of transactions. We didn't even do anything this month xD

![Untitled](https://user-images.githubusercontent.com/40718428/110727317-cf26b080-823c-11eb-94f9-abd51379f478.png)

**Reply by nzjony** [reply]
I haven't seen such an error. It seems that too many tiles are being requested in a given time frame, not that the total number is too high, at least that is what I understand from: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429. Do you have some very intense application that downloads many tiles at once?

Or is the API key potentially shared with some CI bot that is doing some heavy testing?

Can you see in the network panel of your browser if there is a [Retry-After](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) header?

**Reply by gallickgunner** [reply]
I don't think so, I'm just casually zooming in and panning/rotating and I get this. It's a simple project so there's no such intense app or code written to download this many tiles at once. I don't think the API key is shared with any sort of bot we don't have any sort of testing.

Yes there is retry-after header with value 10. I think you were right about the requests exceeding in a given time frame.
![Untitled](https://user-images.githubusercontent.com/40718428/110785050-47689280-828c-11eb-9a94-452f7ff291a2.png)

Any idea what's causing this. I'm just simply moving the camera and the error appears randomly.

**Reply by nzjony** [reply]
@gallickgunner , very strange.

What url are you requesting, is it something like:  https://vector.hereapi.com/v2/vectortiles/core/mc/17/70404/42987/omv ?

**Reply by gallickgunner** [reply]
The same as given above, i.e. `https://vector.hereapi.com/v2/vectortiles/base/mc`

Edit: I think you meant the url mentioned in GET request? It's shown in the error message

**Reply by gallickgunner** [reply]
Here is a test repo. I'm using harpgl+deckgl.
https://github.com/gallickgunner/Deckgl-Experiments/tree/harp+deck

**Reply by nzjony** [reply]
@gallickgunner , the rate limit you are seeing is part of the freemium model. Unfortunately harp.gl doesn't work will with the rate limiting, we have an internal ticket to address this. My hope is that we can fix this at some stage.

**Reply by gallickgunner** [reply]
Dang, nice to see you still responding to this old issue. I actually left my job a few months back so I'm not working on this anymore. Don't even remember If we managed to solve it or not.

**Reply by nzjony** [reply]
@gallickgunner, yeah, was checking the unanswered discussions and noticed this. Ok, hope you found a great new job and thanks anyway for the interest in harp.gl at the time. All the best for your future career!
